src/ContentType/ContentType.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of sensiolabs-de/storyblok-bundle.
+ *
+ * (c) SensioLabs Deutschland <info@sensiolabs.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Storyblok\Bundle\ContentType;
+
+use function Symfony\Component\String\u;
+
+abstract readonly class ContentType implements ContentTypeInterface
+{
+    public static function type(): string
+    {
+        return u(static::class)
+            ->afterLast('\\')
+            ->snake()
+            ->toString();
+    }
```

Comment:
This could throw an error using `namespace App;`, while it is very unlikely, we could double check, that the string is not empty and throw a proper exception, also we could add then:
```suggestion
    /** @return non-empty-string */
    public static function type(): string
    {
        return u(static::class)
            ->afterLast('\\')
            ->snake()
            ->toString();
    }
```
to fix PHPStan

Reply:
This is already in the interface as you defined it. PHPStan wants us to do an assert after toString. And thats what i dont want. Because here its quite obvious that this is never empty

Reply:
what about a file without a namespace extending this class. It is empty then, cause there is no `\` in the namespace

Reply:
Yes then the namespace is empty but it still snake cases the class name. the only thing i dont see it work is an anonymous class. `->afterLast()` does nothing when there is no match. and we have the FQCN here so if there is no namespace it doesn't really matter for this logic

Reply:
But i added an assertation (also if it will never be triggered)